app/models/pacbio/run.rb

```diff
@@ -99,10 +99,19 @@ def update_smrt_link_options(options)
 
     # Returns all the used aliquots
     # @return [Array<Aliquot>] an array of the used aliquots in a run
-    def used_aliquots
+    def aliquots_to_publish_on_run
+      to_publish = []
       plates.flat_map do |plate|
-        plate.wells.flat_map(&:used_aliquots)
+        source_aliquots = plate.wells.flat_map(&:used_aliquots).select do |aliquot|
+          aliquot.source_type == 'Pacbio::Pool'
+        end
+        lib_used_aliquots = source_aliquots.flat_map(&:source).flat_map(&:used_aliquots)
+                                           .select do |aliquot|
+          aliquot.source_type == 'Pacbio::Library'
+        end
+        to_publish.concat(source_aliquots, lib_used_aliquots)
```

Comment:
Additionally, it would be helpful to add some documentation here to clarify what we are trying to accomplish, as it could be confusing for someone picking it up later.